File: 670/MaximumSwap.java
/*
https://leetcode.com/problems/maximum-swap/description/?envType=daily-question&envId=2024-10-17
*/

class Solution {

    public static void swap(StringBuilder str, int left, int right){
        char temp=str.charAt(left);
        str.setCharAt(left, str.charAt(right));
        str.setCharAt(right, temp);
    }

    public int maximumSwap(int num) {
        StringBuilder numToString=new StringBuilder(String.valueOf(num));
        int lengthOfNumString=numToString.length(), resultantMaximumValue=Integer.MIN_VALUE;
        for(int i=0; i<lengthOfNumString; ++i){
            int left=i, right=lengthOfNumString-1;
            while(left<=right){
                StringBuilder stringToCheck=new StringBuilder(numToString);
                swap(stringToCheck, left, right);
                int numToCheck=Integer.parseInt(String.valueOf(stringToCheck));
                resultantMaximumValue=Math.max(resultantMaximumValue, numToCheck);
                right--;
            }
        }
        return resultantMaximumValue;
    }
}

File: 6/670/MaximumSwap.cpp
/*
https://leetcode.com/problems/maximum-swap/description/?envType=daily-question&envId=2024-10-17
*/

class Solution {
public:
    int maximumSwap(int num) {
        string nmstr=to_string(num);
        int sz=nmstr.size(), res=INT_MIN;
        for(int i=0; i<sz; ++i){
            int left=i, right=sz-1;
            while(left<=right){
                string tochk=nmstr;
                char &atlft=tochk[left], &atright=tochk[right];
                swap(atlft, atright);
                int tochknm=stoi(tochk);
                res=max(res, tochknm);
                right--;
            }
        }
        return res;
    }
};